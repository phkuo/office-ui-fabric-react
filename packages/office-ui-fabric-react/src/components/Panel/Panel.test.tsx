import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { PanelBase } from './Panel.base';
import { Panel } from './Panel';
let div: HTMLElement;

const ReactDOM = require('react-dom');

describe('Panel', () => {
  it('renders Panel correctly', () => {
    // Mock createPortal to capture its component hierarchy in snapshot output.
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });

    const component = renderer.create(
      <Panel isOpen={true} headerText="Test Panel">
        <span>Content goes here</span>
      </Panel>
    );

    expect(component.toJSON()).toMatchSnapshot();

    ReactDOM.createPortal.mockClear();
  });

  describe('onClose', () => {
    beforeEach(() => {
      div = document.createElement('div');
    });

    afterEach(() => {
      ReactDOM.unmountComponentAtNode(div);
    });

    it('fires the correct events when closing', done => {
      let dismissedCalled = false;
      let dismissCalled = false;
      const setDismissTrue = (): void => {
        dismissCalled = true;
      };
      const setDismissedTrue = (): void => {
        dismissedCalled = true;
      };

      const panel: PanelBase = ReactDOM.render(
        <PanelBase isOpen={true} onDismiss={setDismissTrue} onDismissed={setDismissedTrue} />,
        div
      ) as any;

      panel.dismiss();

      expect(dismissCalled).toEqual(true);
      expect(dismissedCalled).toEqual(false);

      setTimeout(() => {
        try {
          expect(dismissedCalled).toEqual(true);
          done();
        } catch (e) {
          done(e);
        }
      }, 250);
    });
  });
});
