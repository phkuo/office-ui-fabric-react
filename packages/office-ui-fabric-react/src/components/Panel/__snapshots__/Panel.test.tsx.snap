// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders Panel correctly 1`] = `
<span
  className="ms-layer"
>
  <div
    className=
        ms-Fabric
        ms-Layer-content
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: #333333;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          visibility: visible;
        }
        & button {
          font-family: inherit;
        }
        & input {
          font-family: inherit;
        }
        & textarea {
          font-family: inherit;
        }
        button {
          margin-bottom: 0px;
          margin-left: 0px;
          margin-right: 0px;
          margin-top: 0px;
          overflow: visible;
        }
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDoubleClick={[Function]}
    onDrag={[Function]}
    onDragEnd={[Function]}
    onDragEnter={[Function]}
    onDragExit={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onSubmit={[Function]}
  >
    <div
      aria-labelledby="Panel0-headerText"
      className=


      onKeyDown={[Function]}
      role="dialog"
      style={
        Object {
          "overflowY": undefined,
        }
      }
    >
      <div
        className=
            ms-Panel
            is-open
            ms-Panel--hasCloseButton
            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              bottom: 0px;
              font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
              font-size: 14px;
              font-weight: 400;
              left: 0px;
              pointer-events: none;
              position: absolute;
              right: 0px;
              top: 0px;
            }
        type={1}
      >
        <div
          className=
              ms-Overlay
              {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                animation-duration: 0.267s;
                animation-fill-mode: both;
                animation-name: keyframes from{opacity:0;}to{opacity:1;};
                animation-timing-function: cubic-bezier(.1,.25,.75,.9);
                background-color: rgba(255,255,255,.4);
                bottom: 0px;
                cursor: pointer;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 14px;
                font-weight: 400;
                left: 0px;
                opacity: 1;
                pointer-events: auto;
                position: absolute;
                right: 0px;
                top: 0px;
                transition: opacity 0.367s cubic-bezier(.1,.9,.2,1);
              }
              @media screen and (-ms-high-contrast: active){& {
                border: 1px solid WindowText;
              }
              @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none){& {
                background-color: transparent;
              }
        />
        <div
          className=
              ms-Panel-main
              {
                -webkit-overflow-scrolling: touch;
                animation-duration: 0.367s;
                animation-fill-mode: both;
                animation-name: keyframes from{opacity:0;}to{opacity:1;} keyframes from{transform:translate3d(40px,0,0);}to{transform:translate3d(0,0,0);};
                animation-timing-function: cubic-bezier(.1,.9,.2,1);
                background-color: #ffffff;
                bottom: 0px;
                display: flex;
                flex-direction: column;
                max-height: 100%;
                overflow-x: hidden;
                overflow-y: auto;
                position: absolute;
                right: 0px;
                top: 0px;
                width: 100%;
              }
              @media (min-width: 480px){& {
                border-left: 1px solid #eaeaea;
                border-right: 1px solid #eaeaea;
                box-shadow: 0px 0px 30px 0px rgba(0,0,0,0.2);
                left: auto;
                pointer-events: auto;
                width: 340px;
              }
              $root & {
                pointer-events: auto;
                width: 272px;
              }
              @media (min-width: 480px){$root & {
                width: 340px;
              }
              @supports (-webkit-overflow-scrolling: touch){& {
                max-height: 768px;
              }
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          style={Object {}}
        >
          <div
            className=
                ms-Panel-commands

            data-is-visible={true}
          >
            <div
              className=
                  ms-Panel-navigation
                  {
                    display: flex;
                    height: 44px;
                    justify-content: flex-end;
                    padding-bottom: 0;
                    padding-left: 5px;
                    padding-right: 5px;
                    padding-top: 0;
                  }
            >
              <button
                className=
                    ms-Button
                    ms-Button--icon
                    ms-Panel-closeButton
                    ms-PanelAction-close
                    {
                      -moz-osx-font-smoothing: grayscale;
                      -webkit-font-smoothing: antialiased;
                      background-color: transparent;
                      border-radius: 0px;
                      border: none;
                      box-sizing: border-box;
                      color: #666666;
                      cursor: pointer;
                      display: inline-block;
                      font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                      font-size: 20px;
                      font-weight: 400;
                      height: auto;
                      outline: transparent;
                      padding-bottom: 0;
                      padding-left: 4px;
                      padding-right: 4px;
                      padding-top: 0;
                      position: relative;
                      text-align: center;
                      text-decoration: none;
                      user-select: none;
                      vertical-align: top;
                      width: 44px;
                    }
                    &::-moz-focus-inner {
                      border: 0;
                    }
                    .ms-Fabric--isFocusVisible &:focus:after {
                      border: 1px solid #ffffff;
                      bottom: 0px;
                      content: "";
                      left: 0px;
                      outline: 1px solid #666666;
                      position: absolute;
                      right: 0px;
                      top: 0px;
                      z-index: 1;
                    }
                    @media screen and (-ms-high-contrast: active){.ms-Fabric--isFocusVisible &:focus:after {
                      border: none;
                      bottom: -2px;
                      left: -2px;
                      outline-color: ButtonText;
                      right: -2px;
                      top: -2px;
                    }
                    &:active > * {
                      left: 0px;
                      position: relative;
                      top: 0px;
                    }
                    &:hover {
                      color: #333333;
                    }
                    @media screen and (-ms-high-contrast: active){&:hover {
                      border-color: Highlight;
                      color: Highlight;
                    }
                    &:active {
                      color: #0078d4;
                    }
                data-is-focusable={true}
                data-is-visible={true}
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                type="button"
              >
                <div
                  className=
                      ms-Button-flexContainer
                      {
                        align-items: center;
                        display: flex;
                        flex-wrap: nowrap;
                        height: 100%;
                        justify-content: center;
                      }
                >
                  <i
                    aria-hidden={true}
                    className=
                        ms-Button-icon
                        {
                          -moz-osx-font-smoothing: grayscale;
                          -webkit-font-smoothing: antialiased;
                          display: inline-block;
                          flex-shrink: 0;
                          font-family: "FabricMDL2Icons";
                          font-size: 16px;
                          font-style: normal;
                          font-weight: normal;
                          height: 16px;
                          line-height: 16px;
                          margin-bottom: 0;
                          margin-left: 4px;
                          margin-right: 4px;
                          margin-top: 0;
                          speak: none;
                          text-align: center;
                          vertical-align: middle;
                        }
                    data-icon-name="Cancel"
                    role="presentation"
                  >
                    
                  </i>
                </div>
              </button>
            </div>
          </div>
          <div
            className=
                ms-Panel-contentInner
                {
                  display: flex;
                  flex-direction: column;
                  max-height: 100%;
                  overflow-y: hidden;
                }
                @supports (-webkit-overflow-scrolling: touch){& {
                  max-height: 768px;
                }
          >
            <div
              className=
                  ms-Panel-header
                  {
                    flex-grow: 0;
                    margin-bottom: 14px;
                    margin-left: 0;
                    margin-right: 0;
                    margin-top: 14px;
                    padding-left: 16px;
                    padding-right: 16px;
                  }
                  @media screen and (min-width: 768px){& {
                    padding-left: 32px;
                    padding-right: 32px;
                  }
                  @media screen and (min-width: 1366px){& {
                    padding-left: 40px;
                    padding-right: 40px;
                  }
                  @media (min-width: 1024px){& {
                    margin-top: 30px;
                  }
            >
              <p
                aria-level={2}
                className=
                    ms-Panel-headerText
                    {
                      -moz-osx-font-smoothing: grayscale;
                      -webkit-font-smoothing: antialiased;
                      color: #333333;
                      font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                      font-size: 21px;
                      font-weight: 100;
                      line-height: 32px;
                      margin-bottom: 0px;
                      margin-left: 0px;
                      margin-right: 0px;
                      margin-top: 0px;
                    }
                id="Panel0-headerText"
                role="heading"
              >
                Test Panel
              </p>
            </div>
            <div
              className=
                  ms-Panel-scrollableContent
                  {
                    height: 100%;
                    overflow-y: auto;
                  }
                  @supports (-webkit-overflow-scrolling: touch){& {
                    height: 768px;
                  }
              data-is-scrollable={true}
            >
              <div
                className=
                    ms-Panel-content
                    {
                      margin-bottom: 0px;
                      padding-bottom: 20px;
                      padding-left: 16px;
                      padding-right: 16px;
                    }
                    @media screen and (min-width: 768px){& {
                      padding-left: 32px;
                      padding-right: 32px;
                    }
                    @media screen and (min-width: 1366px){& {
                      padding-left: 40px;
                      padding-right: 40px;
                    }
              >
                <span>
                  Content goes here
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</span>
`;
